### PR TITLE
feat: notify defendant of claim details 1/2 (CMC-1250)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-1250

### Change description ###

New CCD state AWAITING_CLAIM_DETAILS_NOTIFICATION
New CCD event NOTIFY DEFENDANT_OF_CLAIM_DETAILS
New bpmn diagram for notify defendant of claim details, sends an email notification
Set claimNotificationDetailsDate

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sabahirfan — DISMISSED** [review]

File: ccd-definition/CaseEvent/User/UserEvents.json [hunk 1]
```diff
@@ -207,8 +226,8 @@
     "ID": "ADD_OR_AMEND_CLAIM_DOCUMENTS",
     "Name": "Add or amend claim documents",
     "Description": "Add or amend documents attached to the claim",
-    "DisplayOrder":11,
-    "PreConditionState(s)": "PENDING_CASE_ISSUED;AWAITING_CASE_NOTIFICATION",
+    "DisplayOrder":13,
+    "PreConditionState(s)": "PENDING_CASE_ISSUED;AWAITING_CASE_NOTIFICATION;AWAITING_CASE_DETAILS_NOTIFICATION",
```
Comment (ckalista) [inline ccd-definition/CaseEvent/User/UserEvents.json]: 👍

File: ccd-definition/CaseEventToComplexTypes/NotifyDefendantOfClaimDetails.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "Upload",
+    "CaseEventID": "NOTIFY_DEFENDANT_OF_CLAIM_DETAILS",
+    "CaseFieldID": "servedDocumentFiles",
+    "ListElementCode": "medicalReports",
+    "FieldDisplayOrder": 4,
+    "DisplayContext": "OPTIONAL",
+    "Min": 1,
+    "FieldShowCondition": "claimType=\"CLINICAL_NEGLIGENCE\" OR claimType=\"PERSONAL_INJURY\""
```
Comment (ckalista) [inline ccd-definition/CaseEventToComplexTypes/NotifyDefendantOfClaimDetails.json:39]: 👍

File: e2e/tests/solicitorCreatesClaim_test.js [hunk 3]
```diff
@@ -24,6 +24,11 @@ Scenario('Solicitor notifies defendant solicitor of claim', async (I) => {
   await I.see(caseEventMessage('Notify claim'));
 });
 
+// Scenario('Solicitor notifies defendant solicitor of claim detials', async (I) => {
```
Comment (ckalista) [inline e2e/tests/solicitorCreatesClaim_test.js:27]: Any reason?

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            .data(caseDetailsConverter.toMap(updatedCaseData))
+            .build();
+    }
+
+    @Override
+    public Map<String, String> addProperties(CaseData caseData) {
+        return Map.of(
+            CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference(),
+            RESPONDENT_NAME, getPartyNameBasedOnType(caseData.getRespondent1()),
+            ISSUED_ON, formatLocalDate(caseData.getClaimIssuedDate(), DATE)
```
Comment (ckalista) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java:79]: Hmm, is this issue date the correct on here? We will have a follow up story for notification to address this.

File: src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/ParticularsOfClaimValidator.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.reform.unspec.validation.interfaces;
```
Comment (ckalista) [inline src/main/java/uk/gov/hmcts/reform/unspec/validation/interfaces/ParticularsOfClaimValidator.java:1]: Good stuff.

**Review by ckalista — DISMISSED** [review]
Good stuff. 

As discussed since 'CREATED' is what we now call  'AWAITING DEFENDANT RESPONSE'
we could take the opportunity to rename it here (to unconfuse us) - Equally as happy to leave it until we play the state refactoring user story

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java (hunk 4 above)
Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java:79]: not actually sure, the email wanted the claim issue date, but whether that is when the claim was created in ccd or when the claim was shared I am not sure. I guess this is going to be defined by the date refactor ticket?

**Review by dharmendrak — DISMISSED** [review]

**Review by pd-dudkiewicz — DISMISSED** [review]

**Review by sabahirfan — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java (hunk 4 above)
Comment (ckalista) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/camunda/notification/DefendantClaimDetailsNotificationHandler.java:79]: Yeah we will look into this specific issue as part of another ticket.

**Review by ckalista — DISMISSED** [review]

**Review by douglasrice — DISMISSED** [review]

**Review by sabahirfan — DISMISSED** [review]

**Review by douglasrice — APPROVED** [review]

**Pull request merged**